src/components/access-settings/AccessSettings.tsx

```diff
@@ -320,7 +347,7 @@ const AccessSettings: FunctionComponent<{study: Study}> = ({study}) => {
               onClick={() => setIsAddingNewMember(true)}
               variant="contained"
               color="primary"
-              disabled={isAddingNewMember}>
+              disabled={isAddingNewMember || isAccessLoading}>
```

Comment:
Ensure that the user does not click the "Add New Member" button before the members response returns -- otherwise, the "Add New Member" panel is briefly shown then removed by the `updateAccess` call which removes the "Add New Member" panel when the members are updated

Reply:
This component is kind of a useEffect nightmare but I would have to know more to understand if/how it could be improved to make any suggestions.

Reply:
Agreed! I opened [a ticket](https://sagebionetworks.jira.com/browse/DHP-1145) with some ideas for simplifying this component.